sam_view.c

```diff
@@ -817,6 +825,37 @@ static int getLength(char **s)
     return n;
 }
 
+void copy_tag(const char *tag, const bam1_t *rec, kstring_t *linebuf)
+{
+    uint8_t *s = bam_aux_get(rec, tag);
+    if (s) {
+        char aux_type = *s;
+        switch (aux_type) {
+            case 'C':
+            case 'I':
+            case 'S': aux_type = 'I'; break;
+            case 'c':
+            case 's': aux_type = 'i'; break;
+            case 'd': aux_type = 'f'; break;
+        }
+
+        kputc('\t', linebuf);
+        kputsn(tag, 2, linebuf);
+        kputc(':', linebuf);
+        kputc(aux_type=='I'? 'i': aux_type, linebuf);
+        kputc(':', linebuf);
+        switch (aux_type) {
+            case 'H':
+            case 'Z': kputs(bam_aux2Z(s), linebuf); break;
+            case 'i': kputw(bam_aux2i(s), linebuf); break;
+            case 'I': kputuw(bam_aux2i(s), linebuf); break;
+            case 'A': kputc(bam_aux2A(s), linebuf); break;
+            case 'f': kputd(bam_aux2f(s), linebuf); break;
+            default:  kputs("*** Unknown Type ***", linebuf); break;
```

Comment:
> This isn't an appropriate way of returning an error. It ought to be non-void and return -1 or similar.

Ah. I wasn't trying to return an error, and I didn't want to treat this as a fatal error, but just to mark it somehow as 'unhandled' and carry on. Not handling type 'B' was a deliberate decision, but in that case I agree that 'unknown type' is a bit misleading. Perhaps 'unhandled type' would be better.
An alternate approach would be to silently ignore any tag with an unhandled type, or put a warning out to stderr.